### ISSUE TITLE
Tree search not working an intended - Dealing with virtual loss

### ISSUE THREAD

**Issue by evanroberts85** [gcp/leela-zero#782]
The calculation to get the evaluation score for a node when there is virtual loss does not seem to be correct (at least for me).  This results in the tree search being pretty messed up, as it causes the last child node that was visited to almost never be visited in the following play-out, as it is given a lower win_rate score than it should.

Here is my debugging output mid way through the tree search after I had played the opening move as black and asked leela for its response:

```
  **score_without_extra_vloss=0.5618**     
  **score_with_extra_vloss=0.4761**     
  **score_correct=0.5207**     
 Q16 --> VL: 6,   Visits: 32, win: 0.48  psa: 0.23,   puct: 0.06 
 D16 --> VL: 6,   Visits: 33, win: 0.52  psa: 0.23,   puct: 0.05 
  Q4 --> VL: 6,   Visits: 29, win: 0.52  psa: 0.20,   puct: 0.05 
  D4 --> VL: 6,   Visits: 4, win: 0.51  psa: 0.03,   puct: 0.05 
```

The correct score I got from simply dividing get_blackevals() by get_visits() without any static casting. Score_without_extra_vloss is the score if virtual loss is not added to blackeval

Here is the get_eval() code in  UCTNode.cpp in Leela 11.0:
```
float UCTNode::get_eval(int tomove) const {
    // Due to the use of atomic updates and virtual losses, it is
    // possible for the visit count to change underneath us. Make sure
    // to return a consistent result to the caller by caching the values.
    auto virtual_loss = int{m_virtual_loss};
    auto visits = get_visits() + virtual_loss;
    if (visits > 0) {
        auto blackeval = get_blackevals();
        if (tomove == FastBoard::WHITE) {
            blackeval += static_cast<double>(virtual_loss);
        }
        auto score = static_cast<float>(blackeval / (double)visits);
        if (tomove == FastBoard::WHITE) {
            score = 1.0f - score;
        }
        return score;
    } else {
        // If a node has not been visited yet,
        // the eval is that of the parent.
        auto eval = m_init_eval;
        if (tomove == FastBoard::WHITE) {
            eval = 1.0f - eval;
        }
        return eval;
    }
}
```

**Comment by gcp** [comment]
>it causes the last child node that was visited to almost never be visited in the following play-out

This is false:
https://github.com/gcp/leela-zero/blob/47ddf38327b80ec5f2303344d1d9c65e25f5a3fb/src/UCTSearch.cpp#L109

The whole point of virtual loss is to make a being-visited node unattractive to *other* threads. If you break during a search run, you will see intentionally lowered scores.

**Comment by evanroberts85** [comment]
Ah ok.
